src/component/marker/MarkAreaView.ts

```diff
@@ -248,22 +248,19 @@ class MarkAreaView extends MarkerView {
             const points = map(dimPermutations, function (dim) {
                 return getSingleMarkerEndPoint(areaData, idx, dim, seriesModel, api);
             });
+            const xScale = coordSys.getAxis('x').scale;
+            const yScale = coordSys.getAxis('y').scale;
+            const xAxis = xScale.getExtent();
+            const yAxis = yScale.getExtent();
+            const xPoint = [xScale.parse(areaData.get('x0', idx)), xScale.parse(areaData.get('x1', idx))];
+            const yPoint = [yScale.parse(areaData.get('y0', idx)), yScale.parse(areaData.get('y1', idx))];
+            xPoint.sort();
+            yPoint.sort();
```

Comment:
There is a helper to sort an array to asc order. https://github.com/apache/echarts/blob/master/src/util/number.ts#L145

Default compare function in sort will convert the values to strings and use Unicode to compare. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#parameters

Reply:
thanks for helping!